### PR TITLE
Fix run seeds when offline

### PR DESCRIPTION
**Pull request by eugenk**
Seeds depended on an internet connection. With this change they don't:
* Proof statuses and categories have fixtures. They are only used if the download from ontohub.org failed.
* Oops is skipped if either a SocketError (no connection at all) or a RestClient error (e.g. 404 or similar) occurs.

In both cases a message with the error is printed to stdout.

### CODE REVIEW

**Comment by eugenk** [comment]
As requested in the meeting yesterday, I changed the default behaviour to use the local fixtures only. By setting an environment variable (`DOWNLOAD_FIXTURES`), we can choose to use online files from ontohub.org instead. The rebuild script sets this environment variable if the `-d` option is supplied.

Also, `-i` makes the rebuild script only restart invoker and not clean the state.

**Comment by 0robustus1** [comment]
I'd prefer `-r` (for `restart`) instead of `-i` (for `invoker`, i assume?).

**Comment by eugenk** [comment]
Renamed with 51bed36

File: db/seeds/070-oops.rb [hunk 1]
```diff
@@ -1,18 +1,32 @@
+def handle_error(e, reason)
+  puts <<MSG
```
Comment (0robustus1) [inline db/seeds/070-oops.rb]: Please use the `<<-MSG` way and indent the closing `MSG` correctly.

File: db/seeds/070-oops.rb [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
+  Could not seed "oops" because #{reason}.
```
Comment (0robustus1) [inline db/seeds/070-oops.rb]: Indent to the far left corner, or use `squish`.

File: lib/tasks/generate.rake [hunk 3]
```diff
@@ -16,13 +16,22 @@ end
...
   basename = File.basename(filepath)
+  if ENV['DOWNLOAD_FIXTURES']
+    source_url =
+      "https://ontohub.org/repositories/meta/master/download/#{source_file}"
+    temp_file = Tempfile.new([target_file, File.extname(target_file)]).path
+    if system("wget -O #{temp_file} #{source_url}")
+      filepath = temp_file
+    else
+      puts
+        'No connection to ontohub.org. Using local file for the current task.'
```
Comment (0robustus1) [inline lib/tasks/generate.rake]: That won't put the string, but actually prints an empty line (`puts` can take zero arguments).
Either put the string in the same line as `puts` or use Docstrings.

File: script/rebuild-ontohub [hunk 4]
```diff
@@ -34,7 +34,26 @@ execute_or_die() {
 abort_unless_invoker_installed
 initialize_new_invoker_instance
 
-execute_or_die "bundle exec rake elasticsearch:wipe"
-execute_or_die "bundle exec rake db:migrate:clean"
-execute_or_die "redis-cli flushdb"
-execute_or_die "bundle exec rake db:seed"
+for i in "$@"
```
Comment (0robustus1) [inline script/rebuild-ontohub]: please use `; do` ;)

File: script/rebuild-ontohub [hunk 5]
```diff
@@ -34,7 +34,26 @@ execute_or_die() {
...
+    RESTART_INVOKER_ONLY=true
+    ;;
+    *)
+    # unknown option
+    ;;
+  esac
+  shift
+done
+
+if [[ !($RESTART_INVOKER_ONLY) ]]
```
Comment (0robustus1) [inline script/rebuild-ontohub]: please use `; then`

**Comment by 0robustus1** [comment]
:+1:

**Pull request merged**